src/Microsoft.ML.Data/Scorers/SchemaBindablePredictorWrapper.cs

```diff
@@ -426,9 +426,16 @@ private protected override bool SaveAsOnnxCore(OnnxContext ctx, RoleMappedSchema
 
             var mapper = ValueMapper as ISingleCanSaveOnnx;
             Contracts.CheckValue(mapper, nameof(mapper));
-            Contracts.Assert(schema.Feature.HasValue);
             Contracts.Assert(Utils.Size(outputNames) == 3); // Predicted Label, Score and Probablity.
 
+            // Prior doesn't have a feature column and uses the training label column to determine predicted labels
+            if (!schema.Feature.HasValue) {
+                Contracts.Assert(schema.Label.HasValue);
+                var labelColumnName = schema.Label.Value.Name;
+                return mapper.SaveAsOnnx(ctx, outputNames, ctx.GetVariableName(labelColumnName));
+            }
+
+            Contracts.Assert(schema.Feature.HasValue);
```

Comment:
Strictly speaking, the assert is unnecessary here because the if block has a return embedded in it.